File: My Solutions/2022/Day 1/2.py
f = open("in.txt", "r")
ma = 0
mb = 0
mc = 0
now = 0
for line in f.read().split("\n"):
    if line == "":
        if ma <= now:
            mc = mb
            mb = ma
            ma = now
        elif mb <= now:
            mc = mb
            mb = now
        elif mc <= now:
            mc = now
        now = 0
    else:
        now += int(line)
print(ma+mb+mc)

File: My Solutions/2023/Day 2/2.js
const fs = require("fs");

var text = fs.readFileSync("./in.txt").toString();
var total = 0;
for (let line of text.split("\n")) {
    let maxes = {"red":0,"green":0,"blue":0};
    for (let round of line.split(": ")[1].trim().split("; ")) {
        for (let chips of round.split(", ")) {
            if (maxes[chips.split(" ")[1]] < Number(chips.split(" ")[0])) {
                maxes[chips.split(" ")[1]] = Number(chips.split(" ")[0]);
            }
        }
    }
    total += (maxes.red * maxes.green * maxes.blue);
}
console.log(total);